pkg/server/endpoints/eventTracker_test.go

```diff
@@ -0,0 +1,1120 @@
+package endpoints_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/spiffe/spire/pkg/server/endpoints"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPollPeriods(t *testing.T) {
+	for _, tt := range []struct {
+		name         string
+		pollInterval time.Duration
+		pollDuration time.Duration
+
+		expectedPollPeriods uint
+	}{
+		{
+			name:         "polling always polls at least once, even for zero duration",
+			pollInterval: time.Duration(1) * time.Minute,
+			pollDuration: time.Duration(0) * time.Minute,
+
+			expectedPollPeriods: 1,
+		},
+		{
+			name:         "polling always polls at least once, even for negative durations",
+			pollInterval: time.Duration(1) * time.Minute,
+			pollDuration: time.Duration(-10) * time.Minute,
+
+			expectedPollPeriods: 1,
+		},
+		{
+			name:         "minimum poll interval of one second",
+			pollInterval: time.Duration(0) * time.Second,
+			pollDuration: time.Duration(10) * time.Second,
+
+			expectedPollPeriods: 10,
+		},
+		{
+			name:         "minimum poll interval of one second, even for negative intervals",
+			pollInterval: time.Duration(-100) * time.Second,
+			pollDuration: time.Duration(10) * time.Second,
+
+			expectedPollPeriods: 10,
+		},
+		{
+			name:         "polling every minute in two mintues",
+			pollInterval: time.Minute * time.Duration(1),
+			pollDuration: time.Minute * time.Duration(2),
+
+			expectedPollPeriods: 2,
+		},
+		{
+			name:         "polling every minute of an hours",
+			pollInterval: time.Minute * time.Duration(1),
+			pollDuration: time.Hour * time.Duration(1),
+
+			expectedPollPeriods: 60,
+		},
+		{
+			name:         "polling rounds up",
+			pollInterval: time.Minute * time.Duration(3),
+			pollDuration: time.Minute * time.Duration(10),
+
+			expectedPollPeriods: 4,
+		},
+	} {
+		tt := tt
```

Comment:
why is `tt := tt` needed in places?
- I thought go recently got rid of loop var footguns
- while spire test code is a bit mixed on whether it's in the pattern of "define table tests and then the loop" vs pattern here of "define loop and table at same time", even in latter pattern we don't have this assignment

Reply:
This is an oddity of golang and how it's memory model works.  The tt on the left is a "looping" variable and is defined in the `for _, tt := range []struct {` above.  The tt on the right is a capture variable to copy the other tt's value into the block.

Since some modes of running tests involve running them in goroutines, and some modes of goroutine launches can launch multiple tests at a time, if one goroutine advances (due to CPU scheduling) faster than another which is running slow, the right hand side tt will get a new value.  If we didn't capture the value locally, a test might start on one test scenario, and then due to the race condition, finish with checks from another.

Reply:
This is no longer the case as of go1.22 (we're on go1.23).

Reply:
If this is not needed, I can remove it; however, it's commonly mentioned as a requirement to capture the test locally in the block, should the testing framework get altered to use parallel, multiple (possibly race condition creating) goroutines.

Reply:
Ok, I didn't see that this need to capture was removed in the transition to go 1.23.0+  I'm removing it from all my tests (in the next push).

Reply:
looks like baseline code just has references, which we should clean up separately :)